packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -590,6 +711,7 @@ export class Pipeline extends PipelineBase {
       ...this.validateHasStages(),
       ...this.validateStages(),
       ...this.validateArtifacts(),
+      ...this.validateVariables(),
```

Comment:
Considering that elements are added to `variables` later with the `addVariable` method as well as constructor, I defined them in the `validatePipeline` method called in `node.addValidation`.